android/src/main/java/com/pdftron/reactnative/views/DocumentView.java

```diff
@@ -3132,12 +3134,56 @@ public void importAnnotations(String xfdf, boolean replace) throws PDFNetExcepti
                     pdfDoc.fdfMerge(fdfDoc);
                 }
                 pdfViewCtrl.update(true);
+
             } finally {
                 if (shouldUnlock) {
                     pdfViewCtrl.docUnlock();
                 }
             }
         }
+        try {
+            return getAnnotationsFromXFDF(xfdf);
```

Comment:
you only want to return this when the merge is successful, so you want to have this line right after pdfviewctrl.update(true)

Reply:
what about for collaboration mode?

Reply:
return this after "mCollabManager.importAnnotations(xfdf, false);"

Reply:
you want to maintain the code readability, so return when needed is important